### ISSUE TITLE
How to run a training with tune for a specific number of steps or episodes

### ISSUE THREAD

**Issue by armando-fandango** [ray-project/ray#15662]
I would like to train a RL agent for a specific number of steps or episodes, say 100 episodes or 10000 steps.

If I train for # of episodes, then the training would proceed for each episode until done, or until max_steps_per_episode aka episode_length is reached.

If I train for # of steps, then that would be total number of steps, irrespective of how many steps were run for each episode.

What do I need to attach to the following code to be able to do that please ?

config = ppo.DEFAULT_CONFIG.copy()
config["log_level"] = "WARN"
config["framework"] = "torch"
config["env"]="Taxi-v3"

ray.shutdown()
ray.init(ignore_reinit_error=True)

result = tune.run(
    ppo.PPOTrainer,
    config=config,
    local_dir = './ray_results/'
)

**Comment by mvindiola1** [comment]
Hi @armando-fandango,

You want to pass a dictionary to the kwarg "stop". You can find more on stopping here https://docs.ray.io/en/master/tune/user-guide.html#stopping-trials.

```python
result = tune.run(
ppo.PPOTrainer,
config=config,
local_dir = './ray_results/', 
stop={"timesteps_total": 10000,  "episodes_total":100} 
)

By the way if you have anymore questions we are happy to help. The ray team preferes if questions like this see posted at https://discuss.ray.io/

**Comment by armando-fandango** [comment]
@mvindiola1 Excellent and thank you. The reason I posted here, was because I think its a bug.

On trying both options, one by one, this the output I get:

1. with "timesteps_total": 50 in the output I get:

timesteps_total: 4000

I thought maybe it needs a minimum of 4000 timesteps, so I set it to 4050 in stop:

timesteps_total: 8000

So why is it truncating timesteps_total to a multiple of 4000?

2. with episodes_total, I observed the same thing that episodes were not being executed for the number of episodes asked, but in multiples of 20.
